### DISCUSSION TITLE
How to set firefox profile path?

### DISCUSSION THREAD

**Discussion by bshsbr** [seleniumbase/SeleniumBase#3316] [Q&A] [answered]
user_data_dir only works on chrome,
but what about firefox?
how to set firefox profile path?
`with SB(browser="firefox", user_data_dir=r"Profiles\ultibot") as sb:`

**Comment by mdmintz** [comment, marked as answer]
If there's a way to set a Firefox Profile, then it would be done via `firefox_arg` or `firefox_pref`:

```
  firefox_arg=FIREFOX_ARG,
        Add a Firefox argument for Firefox browser runs.
        Format: A comma-separated list of Firefox args.
  firefox_pref=FIREFOX_PREF,
        Set a Firefox preference:value combination.
        Format: A comma-separated list of pref:value items.
        Example usage:
            firefox_pref="browser.formfill.enable:True"
        Boolean and integer values to the right of the ":"
        will be automatically converted into proper format.
        If there's no ":" in the string, then True is used.
```

https://github.com/seleniumbase/SeleniumBase/blob/c909b1aae04172be680d75543ea9acfa2521af54/seleniumbase/plugins/sb_manager.py#L195

https://github.com/seleniumbase/SeleniumBase/blob/c909b1aae04172be680d75543ea9acfa2521af54/seleniumbase/plugins/sb_manager.py#L196

**Reply by bshsbr** [reply]
Thanks, it helped!
I used this syntax:
`with SB(browser="firefox", firefox_arg=fr"--profile={profile_path}") as driver:`
but now I'm getting read timeout error, for some reason geckodriver and firefox can't connect to each other:
```
HTTPConnectionPool(host='localhost', port=9106): Read timed out. (read timeout=120)
TimeoutError: timed out
```
